src/stages/stage3_events.c

```diff
@@ -741,939 +81,689 @@ Boss* stage3_spawn_scuttle(cmplx pos) {
+		g = 1.0;
+	} else {
+		r = 1.0;
+		g = 0.3;
+	}
 
-			PARTICLE(
-				.proto = pp_blast,
-				.pos = p->pos,
-				.color = c,
-				.timeout = 35 - 5 * frand(),
-				.draw_rule = GrowFade,
-				.args = { 0, 1 + 0.5 * frand() },
-				.angle = M_PI * 2 * frand(),
-			);
-
-			// FIXME: better sound
-			play_sound("enemydeath");
-			play_sound("shot1");
-			play_sound("shot3");
-		}
+	int cnt = difficulty_value(24, 20, 16, 12);
+	for(int i = 0; i < cnt; ++i) {
+		double a = (M_PI * 2.0 * i) / cnt;
+		cmplx dir = cdir(a);
+		log_debug("shot_type: %d", ARGS.shot_type);
 
-		return ACTION_DESTROY;
+		PROJECTILE(
+				.proto = ARGS.shot_type ? pp_ball : pp_rice,
+				.pos = e->pos,
+				.color = RGB(r, g, 1.0),
+				.move = move_asymptotic_simple(1.5 * dir, 10 - 10 * psin(2 * a + M_PI/2)),
+				);
 	}
 
-	Laser *laser = (Laser*)REF(p->args[0]);
+}
 
-	if(!laser)
-		return ACTION_DESTROY;
+TASK(burst_swirl, { cmplx pos; cmplx dir; int shot_type; }) {
+	// swirls - fly in, explode when killed or after a preset time
+	Enemy *e = TASK_BIND_UNBOXED(create_enemy1c(ARGS.pos, 200, Swirl, NULL, 0));
 
-	p->pos = laser->prule(laser, time);
+	// drop items when dead
+	INVOKE_TASK_WHEN(&e->events.killed, common_drop_items, &e->pos, {
+			.points = 1,
+			.power = 1,
+			});
 
-	return 1;
-}
+	// die after a preset time...
+	INVOKE_TASK_DELAYED(60, destroy_enemy, ENT_BOX(e));
+	// ... and explode when they die
+	log_debug("shot_type: %d", ARGS.shot_type);
+	INVOKE_TASK_WHEN(&e->events.killed, death_burst, ENT_BOX(e), ARGS.shot_type);
 
-DEPRECATED_DRAW_RULE
-static void wriggle_slave_part_draw(Projectile *p, int t, ProjDrawRuleArgs args) {
-	float b = 1 - t / (double)p->timeout;
-	r_mat_mv_push();
-	r_mat_mv_translate(creal(p->pos), cimag(p->pos), 0);
-	ProjDrawCore(p, color_mul_scalar(COLOR_COPY(&p->color), b));
-	r_mat_mv_pop();
+	// define what direction they should fly in
+	e->move = move_linear(ARGS.dir);
 }
 
-static int wriggle_spell_slave(Enemy *e, int time) {
-	TIMER(&time)
+TASK(burst_swirls, { int count; int interval; int shot_type; }) {
+	for(int i = 0; i < ARGS.count; ++i) {
+		tsrand_fill(2);
+		// spawn in a "pitchfork" pattern
+		log_debug("shot_type: %d", ARGS.shot_type);
+		INVOKE_SUBTASK(burst_swirl, VIEWPORT_W/2 + 20 * anfrand(0) + (VIEWPORT_H/4 + 20 * anfrand(1))*I, 3 * (I + sin(M_PI*global.frames/15.0)), ARGS.shot_type);
+		WAIT(ARGS.interval);
+	}
+}
 
-	float angle = e->args[2] * (time / 70.0 + e->args[1]);
-	cmplx dir = cexp(I*angle);
-	Boss *boss = (Boss*)REF(e->args[0]);
+// side swirls
+// typically move across a stage in a drive-by fashion
 
-	if(!boss)
-		return ACTION_DESTROY;
+TASK(side_swirl_move, { BoxedEnemy e; cmplx p0; cmplx p1; cmplx p2; } ) {
+	Enemy *e = TASK_BIND(ARGS.e);
+	cmplx p0 = ARGS.p0;
+	cmplx p1 = ARGS.p1;
+	cmplx p2 = ARGS.p2;
 
-	AT(EVENT_BIRTH) {
-		e->ent.draw_layer = LAYER_BULLET - 1;
+	for (;;) {
+		e->move = move_linear(p0 + p1 * creal(p2));
+		p2 = creal(p2) * cimag(p2) + I * cimag(p2);
+		YIELD;
 	}
+}
```

Comment:
Wouldn't blame you for not figuring out what the hell is going on here; I don't quite get it myself :)
Some kind of accelerated motion..? It can probably be replaced with a simple `move_accelerated()` or so (too tired to figure that out right now).

In any case, non-descriptive parameters like that aren't nice. It's fine if it's temporary, but in that case a `TODO` comment is appreciated.

Reply:
It's an arc that moves in three directions over time. It arcs downwards and to the right of the screen, and then curves downwards on the ride side of the screen. I think having two separate `move` statements separated by a `WAIT(x)` would make it do that, at least to some degree.